src/lib.nr

```diff
@@ -30,22 +33,31 @@ trait BigNumParamsTrait<N> {
      * @brief modulus: all BigNum operations are evaluated modulo this value
      **/
     fn modulus() -> [Field; N];
+
     /**
      * @brief double_modulus: used when performing negations and subtractions
      **/
     fn double_modulus() -> [Field; N];
+
     /**
      * @brief redc_param used for __barrett_reduction. See https://en.wikipedia.org/wiki/Barrett_reduction
      **/
     fn redc_param() -> [Field; N];
+
     /**
      * @brief k used for __barrett_reduction. Should be at least modulus_bits() + 1
      **/
     fn k() -> u64;
+
     /**
      * @brief modulus_bits = log2(modulus) rounded up
      **/
     fn modulus_bits() -> u64;
+
+    /**
+     * @brief efficient multiplication of 2 N-length arrays. To use in constrained fn
+     **/
+    fn mult(a: [Field; N], b: [Field; N]) -> ArrayX<Field, N, 2>;
```

Comment:
hmm good point.. however then it might be a bit more cumbersome for a user because they would have to implement 2 separate traits. Maybe a renaming would be more user friendly?

Reply:
we'll adjust this later